src/menu/menu.c

```diff
@@ -908,6 +910,117 @@ menu_hide_submenu(struct server *server, const char *id)
+ *
+ * This will look at workspaces and produce a menu
+ * with the workspace names that can be used with
+ * SendToDesktop, left/right options are included.
+ */
+void
+create_client_send_to_menu(struct server *server)
+{
+	struct menu *menu = menu_get_by_id(server,
+			"client-send-to-menu");
+
+	if (menu) {
+		struct menuitem *item, *next;
+		wl_list_for_each_safe(item, next, &menu->menuitems, link) {
+			item_destroy(item);
+		}
+	} else {
+		menu = menu_create(server, "client-send-to-menu", "Send to...");
+	}
+
+	menu->size.height = 0;
+
+	struct workspace *workspace;
+
+	wl_list_init(&menu->menuitems);
+	wl_list_for_each(workspace, &server->workspaces, link) {
+		if (workspace == server->workspace_current) {
+			char *tmp_buf = strdup_printf(">%s<", workspace->name);
+			current_item = item_create(menu, tmp_buf, /*show arrow*/ false);
+			free(tmp_buf);
+		} else {
+			current_item = item_create(menu, workspace->name, /*show arrow*/ false);
+		}
+		fill_item("name.action", "SendToDesktop");
+		fill_item("to.action", workspace->name);
+	}
+
+	current_item = separator_create(menu, "");
+	current_item = item_create(menu, "left", /*show arrow*/ false);
+	fill_item("name.action", "SendToDesktop");
+	fill_item("to.action", "left");
+	current_item = item_create(menu, "right", /*show arrow*/ false);
+	fill_item("name.action", "SendToDesktop");
+	fill_item("to.action", "right");
+	menu_update_width(menu);
+	wlr_scene_node_set_enabled(&menu->scene_tree->node, false);
+}
+
+/*
+ * This is client-list-combined-menu
+ * an internal menu similar to root-menu and client-menu
+ *
+ * This will look at workspaces and produce a menu
+ * with the workspace name as a separator label
+ * and the titles of the view, if any, below each workspace name.
+ */
+void
+create_client_list_combined_menu(struct server *server)
+{
+	struct menu *menu = menu_get_by_id(server,
+			"client-list-combined-menu");
+
+	if (menu) {
+		struct menuitem *item, *next;
+		wl_list_for_each_safe(item, next, &menu->menuitems, link) {
+			item_destroy(item);
+		}
+	} else {
+		menu = menu_create(server, "client-list-combined-menu", "Clients...");
+	}
+
+	menu->size.height = 0;
+
+		struct workspace *workspace;
+		struct view *view;
+		struct buf buffer = BUF_INIT;
+
+		wl_list_init(&menu->menuitems);
+		wl_list_for_each(workspace, &server->workspaces, link) {
+		buf_add_fmt(&buffer, workspace == server->workspace_current ? ">%s<" : "%s",
+				workspace->name);
+		current_item = separator_create(menu, buffer.data);
+		buf_clear(&buffer);
+
+		wl_list_for_each(view, &server->views, link) {
+			if (view->workspace->name == workspace->name) {
+				buf_add(&buffer, view_get_string_prop(view, "title"));
+
+				current_item = item_create(menu, buffer.data, false);
+				current_item->id = xstrdup(menu->id);
+				current_item->client_list_view = view;
+				fill_item("name.action", "GoToDesktop");
+				fill_item("to.action", workspace->name);
```

Comment:
I don't think we need this action, at least from memory focusing a window on another workspace should switch there automatically.

Reply:
Just tested, and works with just focus/raise, at least in my limited testing.